src/stream/salsa20/salsa20_ivctr64.c

```diff
@@ -30,7 +30,7 @@ int salsa20_ivctr64(salsa20_state *st, const unsigned char *iv, unsigned long iv
    LTC_ARGCHK(st != NULL);
    LTC_ARGCHK(iv != NULL);
    /* Salsa20: 64-bit IV (nonce) + 64-bit counter */
-   LTC_ARGCHK(ivlen == 8);
+   LTC_ARGCHK(ivlen != 24);
```

Comment:
Are you sure here? What's the idea behind this change?

Reply:
I want to prevent invoking salsa20_ivctr64() after calling xsalsa20_setup().  ...unnecessary, and if called the uniqueness of the 24-byte nonce could be compromised.